src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/hiddenfield/InputHiddenFieldInnerRecordsImplicitlyStatic.java

```diff
@@ -0,0 +1,54 @@
+/*
+HiddenField
+ignoreConstructorParameter = true
+ignoreSetter = true
+setterCanReturnItsClass = true
+tokens = (default)VARIABLE_DEF, PARAMETER_DEF, PATTERN_VARIABLE_DEF, LAMBDA, RECORD_COMPONENT_DEF
+
+
+*/
+
+//non-compiled with javac: Compilable with Java14
+package com.puppycrawl.tools.checkstyle.checks.coding.hiddenfield;
+
+import java.time.Clock;
+import java.time.Instant;
+
+public class InputHiddenFieldInnerRecordsImplicitlyStatic {
+}
+
+class Scratch {
+
+    private final Clock clock;
+
+    Scratch(Clock clock) {
+        this.clock = clock;
+    }
+
+    public record State(String token, Instant expiresAt) {
+        static int pointer = 0;
+
+        public boolean isFresh(final Clock clock) { // unexpex
+            return Instant.now(clock).isBefore(expiresAt);
```

Comment:
Instead of such strange comment, can you put a verbose comment that explains why this is not violation.